TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1240,42 +1312,86 @@ def exterior_boundaries(feature_layers, zoom,
 
     features = layer['features']
 
+    # this exists to enable a dirty hack to try and work
+    # around duplicate geometries in the database. this
+    # happens when a multipolygon relation can't
+    # supersede a member way because the way contains tags
+    # which aren't present on the relation. working around
+    # this by calling "union" on geometries proved to be
+    # too expensive (~3x current), so this hack looks at
+    # the way_area of each object, and uses that as a
+    # proxy for identity. it's not perfect, but the chance
+    # that there are two overlapping polygons of exactly
+    # the same size must be pretty small. however, the
+    # STRTree we're using as a spatial index doesn't
+    # directly support setting attributes on the indexed
+    # geometries, so this class exists to carry the area
+    # attribute through the index to the point where we
+    # want to use it.
+    class geom_with_area:
+        def __init__(self, geom, area):
+            self.geom = geom
+            self.area = area
+            self._geom = geom._geom
```

Comment:
Ah sorry, I missed that. In that case we'd have to do something like:

    geom_with_area(snapped, props.get('area'), snapped._geom)

Which seems like it's not as worth it. Although we only instantiate these objects in a single place, so maybe it doesn't matter much.

Either way is :+1: to me.